proposals/hypermedia-control-2/README.md

```diff
@@ -0,0 +1,424 @@
+{
+  "@context": "https://www.w3.org/2019/wot/td/v1",
+  "id": "urn:ex:thing",
+  "actions": {
+    "fade": {
+      "input": {
+        "type": "number",
+        "description": "duration in ms"
+      },
+      "output":{
+        "type":"number",
+        "description": "brightness value after fade operation"
+      },
+      "forms": [
+        {
+          "href": "/fade",
+          "op": "invokeaction",
+          "htv:methodName": "POST"
+        }
+      ]
+    }
+  }
+}
+```
+
+Note that this TD is respecting the current standard.
+**Currently, the output would be expected as the response to the `POST /fade` request, i.e. the response of `invokeaction`.**
+
+## Hypermedia Use Case Example
+
+Differently, we want an interface that allows Consumers to invoke a `fade` action of the Lamp Thing that can be additionally queried, updated or cancelled.
+The meaning of these are:
+
+- Querying: Getting information about how the action is ongoing. In our case, it **can** be the current brightness value; status message like `"pending"`, `"ongoing"`, `"finished"` or something else.
+- Updating: For an ongoing or pending action, the Consumer change the desired outcome. In our case, it could be adding time or changing the total time. We can also imagine changing the fade speed etc.
+- Cancelling: For an ongoing or pending action, the Consumer cancel the action. This can be before the action has started or when it is just about to start and can thus require a response to explain the current state.
+
+We can thus imagine the following interface for the Consumer like proposed by @vcharpenay
+
+```
+POST /fade
+GET /fade/{id}
+PUT /fade/{id}
+DELETE /fade/{id}
+```
+
+OR the `href` for hypermedia-related requests being static.
+
+```
+POST /fade
+GET /fade/ongoing
+PUT /fade/ongoing
+DELETE /fade/ongoing
+```
+
+Here is a sequence diagram summarizing possible interactions **that does not show the responses of the Thing**:
+
+![Fade sequence diagram](fade-noResp.png)
+
+With the current TD specification, a standard-compliant Consumer would be able to perform the first request but no subsequent request on the created resource (i.e. `/fade/1` OR  `/fade/ongoing`).
+
+First, we want to **start** by describing that such requests can be sent to the Thing by adding new WoT operation keywords.
+
+Thus, we can have the following TD that is consumed by the Consumer:
+
+```jsonc
+{
+  "@context": "https://www.w3.org/2019/wot/td/v1",
+  "id": "urn:ex:thing",
+  "actions": {
+    "fade": {
+      "input": {
+        "type": "number",
+        "description": "duration in ms"
+      },
+      "output":{
+        "type":"number",
+        "description": "brightness value after fade operation"
+      },
+      "forms": [
+        {
+          "href": "/fade",
+          "op": "invokeaction",
+          "htv:methodName": "POST"
+        },
+        {
+          "href": "/fade/{id}", //OR /fade/ongoing
+          "op": "queryaction",
+          "htv:methodName": "GET"
```

Comment:
Specifying `contentType` for the sake of a clear example would be nice.

Reply:
Fixed this :)